src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -2415,7 +2415,7 @@ private string AppendNumericValueToKey(string key, long val)
             _keyBuilder.Length = 0;
             _keyBuilder.Append(key);
             _keyBuilder[_keyBuilder.Length - 1] = '/';
-            _keyBuilder.AppendFormat(CultureInfo.InvariantCulture, "D19", val);
+            _keyBuilder.AppendFormat(CultureInfo.InvariantCulture, "{0:D19}", val);
```

Comment:
Yes, that give us a number like `0000000000000000001`, which gives us lexical sorting.